### DISCUSSION TITLE
Branch Name Pattern

### DISCUSSION THREAD

**Discussion by Sirisha-V** [community/community#68402] [Apps, API and Webhooks]
### Select Topic Area

General

### Body

Is it possible to update the branch name pattern of a repo and update the branch protection rules, using REST API calls?

**Comment by ShakthiW** [comment]
Hi there,

Yes,

To update the branch name pattern of a repo, you can use the following PUT request:

```
PUT /repos/{owner}/{repo}/branches/protection/branch-name-pattern
{
  "branch_name_pattern": "{new_branch_name_pattern}"
}
```

To update the branch protection rules for a repo, you can use the following PUT request:

```
PUT /repos/{owner}/{repo}/branches/protection/{branch}
{
  // Branch protection rules
}
```

For example, to update the branch protection rules for the `main` branch of the `my-repo` repository, you would use the following PUT request:

```
PUT /repos/my-username/my-repo/branches/protection/main
{
  "required_status_checks": {
    "strict": true,
    "contexts": ["continuous-integration/travis-ci"]
  }
}
```

You can find more information about the REST API for branch protection rules in the GitHub documentation: https://docs.github.com/en/rest/branches/branch-protection: https://docs.github.com/en/rest/branches/branch-protection

I hope this helps!

**Reply by Sirisha-V** [reply]
Hi Thanks for responding 
/repos/{owner}/{repo}/branches/protection/branch-name-pattern
I am unable to access this URL. As its 404 not found can you let me know what I am missing out

**Comment by ShakthiW** [comment]
I'm glad to help. 

That URL should work, but it's important to note that it is a placeholder. You need to replace the `{owner}` and `{repo}` placeholders with the actual name of the repository owner and repository name.

For example, if you wanted to update the branch name pattern of the `my-repo` repository owned by the user `my-username`, you would use the following URL:

```
https://api.github.com/repos/my-username/my-repo/branches/protection/branch-name-pattern
```

**Reply by Sirisha-V** [reply]
Hi, yes I get that. 
The URL I am hitting is 
https://github.myorg.com/repos/test-user/Java/branches/protection/branch-name-pattern
But I am getting 422 error 
422Unprocessable Entity
Can you please help me figure out what mistake I am doing.
